source/server-discovery-and-monitoring/tests/integration/closeInUse-pool-clear.yml

```diff
@@ -0,0 +1,108 @@
+# Test connection pool clear(closeInUse:true).
+runOn:
+    # This test requires failCommand appName, and
+    # failCommand blockConnection with closeConnection:true (SERVER-53512).
+    - minServerVersion: "4.9"
+
+database_name: &database_name "sdam-tests"
+collection_name: &collection_name "closeInUse"
+
+data: []
+
+tests:
+  - description: pool clear close in use true generates expected events
+    useMultipleMongoses: false,
+    clientOptions:
+      connectTimeoutMS: 500, 
+      heartbeatFrequencyMS: 500
+      appname: "closeInUse"
+      retryReads: false
+      minPoolSize: 0
+    operations:
+      - name: insertMany
+        object: collection
+        arguments:
+          documents:
+            - _id: 1
+      - name: startThread
+        object: testRunner
+        arguments:
+          name: thread1
+      # simulate a long-running query
+      - name: runOnThread
+        object: testRunner
+        arguments:
+          name: thread1
+          operation:
+            name: find
+            object: collection
+            arguments:
+              filter: { "$where": "sleep(5000) || true" }
+      - name: configureFailPoint
+        object: testRunner
+        arguments:
+          failPoint:
+            configureFailPoint: failCommand
+            mode: { times: 1 }
+            data:
+                failCommands: ["hello"]
+                blockConnection: true
+                blockTimeMS: 1000
+                appName: closeInUse
+      - name: waitForEvent
+        object: testRunner
+        arguments:
+          event: ConnectionClosed
+          count: 1
+      - name: waitForEvent
+        object: testRunner
+        arguments:
+          event: PoolClearedEvent
+          count: 1
+      - name: waitForEvent
+        object: testRunner
+        arguments:
+          event: ConnectionCheckedInEvent
+          count: 2
+      - name: assertEventCount
+        object: testRunner
+        arguments:
+          event: PoolClearedEvent
+          count: 1
+      # ensure connection is closed
+      - name: assertEventCount
+        object: testRunner
+        arguments:
+          event: ConnectionClosed
+          count: 1 
+      # ensure there are corresponding checkOut/checkIn events
+      - name: assertEventCount
+        object: testRunner
+        arguments:
+          event: ConnectionCheckedOutEvent
+          count: 2
+      - name: assertEventCount
+        object: testRunner
+        arguments:
+          event: ConnectionCheckedInEvent
+          count: 2   
+    expectations:
+      - command_started_event:
+          command:
+            insert: *collection_name
+            documents:
+              - _id: 1
+          command_name: insert
+          database_name: *database_name
+      - command_started_event:
```

Comment:
can we add a `CommandFailedEvent` assertion here?

Reply:
I would not do it for events asserting since SDAM test runner assumes that we validate only CommandStartedEvent, so to make it happen we would need updating a number of unrelated tests. I added this check via: `waitForEvent => CommandFailedEvent`